### PR TITLE
fix(secgroup): fix a bug in security group

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. fix a bug in security group
2. update sdk version to V1 in test file
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
test in cn-north-4

make testacc TEST='./huaweicloud' TESTARGS='-run TestAccNetworkingV3SecGroup_noDefaultRules'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccNetworkingV3SecGroup_noDefaultRules -timeout 360m -parallel 4
=== RUN   TestAccNetworkingV3SecGroup_noDefaultRules
=== PAUSE TestAccNetworkingV3SecGroup_noDefaultRules
=== CONT  TestAccNetworkingV3SecGroup_noDefaultRules
--- PASS: TestAccNetworkingV3SecGroup_noDefaultRules (22.12s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       22.231s

make testacc TEST='./huaweicloud' TESTARGS='-run TestAccNetworkingSecGroupRule_basic'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccNetworkingSecGroupRule_basic -timeout 360m -parallel 4
=== RUN   TestAccNetworkingSecGroupRule_basic
=== PAUSE TestAccNetworkingSecGroupRule_basic
=== CONT  TestAccNetworkingSecGroupRule_basic
--- PASS: TestAccNetworkingSecGroupRule_basic (36.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       36.292s

test in ap-southeast-2

make testacc TEST='./huaweicloud' TESTARGS='-run TestAccNetworkingV3SecGroup_noDefaultRules'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccNetworkingV3SecGroup_noDefaultRules -timeout 360m -parallel 4
=== RUN   TestAccNetworkingV3SecGroup_noDefaultRules
=== PAUSE TestAccNetworkingV3SecGroup_noDefaultRules
=== CONT  TestAccNetworkingV3SecGroup_noDefaultRules
--- PASS: TestAccNetworkingV3SecGroup_noDefaultRules (22.11s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       22.183s

make testacc TEST='./huaweicloud' TESTARGS='-run TestAccNetworkingSecGroupRule_oldPorts'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccNetworkingSecGroupRule_oldPorts -timeout 360m -parallel 4
=== RUN   TestAccNetworkingSecGroupRule_oldPorts
=== PAUSE TestAccNetworkingSecGroupRule_oldPorts
=== CONT  TestAccNetworkingSecGroupRule_oldPorts
--- PASS: TestAccNetworkingSecGroupRule_oldPorts (36.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       36.887s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_networking_secgroup_test.go [hunk 1]
```diff
@@ -8,13 +8,13 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 
-	"github.com/chnsz/golangsdk/openstack/networking/v3/security/groups"
+	"github.com/chnsz/golangsdk/openstack/networking/v1/security/securitygroups"
```
Comment (hmbig2) [inline huaweicloud/resource_huaweicloud_networking_secgroup_test.go:11]: maybe you can just add an alias `group` next time

Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_networking_secgroup_test.go:11]: ok

File: huaweicloud/data_source_huaweicloud_networking_secgroup.go [hunk 2]
```diff
@@ -166,18 +166,16 @@ func dataSourceNetworkingSecGroupReadV1(_ context.Context, d *schema.ResourceDat
...
+
 	logp.Printf("[DEBUG] The retrieved rules list is: %v", rules)
 
 	mErr := multierror.Append(nil,
 		d.Set("region", region),
 		d.Set("name", resp.Name),
 		d.Set("description", resp.Description),
 		d.Set("enterprise_project_id", resp.EnterpriseProjectId),
-		d.Set("rules", rules),
+		d.Set("rules", flattenSecurityGroupRulesV1(&resp)),
```
Comment (niuzhenguo) [inline huaweicloud/data_source_huaweicloud_networking_secgroup.go]: s/flattenSecurityGroupRulesV1/rules

Comment (Jason-Zhang9309) [reply huaweicloud/data_source_huaweicloud_networking_secgroup.go]: sure, fixed

File: huaweicloud/data_source_huaweicloud_networking_secgroup.go [hunk 3]
```diff
@@ -166,10 +166,8 @@ func dataSourceNetworkingSecGroupReadV1(_ context.Context, d *schema.ResourceDat
 	resp := filterGroups[0].(v1groups.SecurityGroup)
 	d.SetId(resp.ID)
 
-	rules, err := flattenSecurityGroupRulesV1(&resp)
-	if err != nil {
-		return diag.FromErr(err)
-	}
+	rules := flattenSecurityGroupRulesV1(&resp)
+
```
Comment (Lance52259) [inline huaweicloud/data_source_huaweicloud_networking_secgroup.go]: Lines 169 and 171 are logically tight, suggest to remove line 170.

Comment (Jason-Zhang9309) [reply huaweicloud/data_source_huaweicloud_networking_secgroup.go]: ok

**Pull request merged**